backend/benchmark.py

```
@@ -117,55 +212,106 @@ def call_rag(query: str, thread_id: str):
 
     def run_metric(args):
+        """Run one metric with up to 3 retries; return (name, result) or (name, exception)."""
         name, fn, kwargs = args
+
         for attempt in range(3):
             try:
                 loop = asyncio.new_event_loop()
                 asyncio.set_event_loop(loop)
                 return name, fn(**kwargs)
+
             except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)